cpp/tests/streaming/test_spillable_messages.cpp

```diff
@@ -172,3 +181,68 @@ TEST_F(StreamingSpillableMessages, MultiThreadedRandomInsertSpillExtract) {
     EXPECT_EQ(extracted_sum.load(), total_msgs);
     EXPECT_TRUE(msgs.get_content_descriptions().empty());
 }
+
+TEST_F(StreamingSpillableMessages, SpillInFlightMessages) {
+    constexpr int num_producer_producer_pairs = 16;
+    constexpr int msgs_per_producer = 100;
+    std::vector<Node> nodes;
+    std::atomic<std::size_t> spilled_expect{0};
+    std::atomic<std::size_t> spilled_got{0};
```

Comment:
It seems like this is only testing the case where no spilling occurs, am I understanding this right? Do we want to have a test case in which spilling actually occurs too?